enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Baremetal-Keepalived-Unicast
+authors:
+  - "@bcrochet"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-02-21
+last-updated: 2020-02-21
+status: implementable
+---
+
+# baremetal-keepalived-unicast
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Baremetal platform uses keepalived to move VIPs between the bootstrap host and the
+masters, then between the masters and the workers. This worked via multicast, but not
+all environments have multicast available. This proposal seeks to rememdy that.
+
+## Motivation
+
+Currently, the Baremetal platform utilizes keepalived to maintain the VIPs necessary
+for deployment and operation. It presently does so via multicast. This requires a
+minimal amount of configuration, as the multicast peers do not need to be specified.
+However, there are likely to be environments where multicast has been disabled because
+of network policy. In these cases, it is necessary to utilized unicast addresses within
```

Comment:
am I right to assume that multicast also require a L2 network connectivity, which in turn means you can't:

1.  have nodes on different vlans (e.g. distribute your masters across failure domains)
2.  run multiple clusters on the same vlan

Reply:
The multicast advertisements already include the cluster name so it's safe to run multiple clusters on the same network.  Any advertisements coming from other clusters would be ignored.

It's possible to set up multicast forwarding between different networks.  I have not seen anyone document an example for how to do so, though.

Reply:
Even if the forwarding is possible, it would be another requirement on the customer, isn't it?

Reply:
Yes.  I wasn't trying to imply that it's great, just answering ohad's question about what is possible